### DISCUSSION TITLE
What is wrong with my AWSBatch cromwell.conf values?

### DISCUSSION THREAD

**Discussion by PeterT2020** [broadinstitute/cromwell#7493] [Q&A] [answered]
Hey there, I'm getting weird issues with my conf. When submitting to cromwell version 59 I see this:
2024-08-15 00:45:56,656  WARN  - Unrecognized configuration key(s) for AwsBatch: auth, numCreateDefinitionAttempts, numSubmitAttempts, default-runtime-attributes.scriptBucketName

So what am I doing wrong here? Do I just absolutely need to upgrade to a newer version of cromwell in order for this to work or is something else going on here?
 
Here is my conf, with the sensitive values replaced:
include required(classpath("application"))

```
aws {
  application-name = "cromwell"
  auths = [
    {
      name = "default"
      scheme = "default"
    }
  ]
  region = "us-east-2"
}

engine {
  filesystems {
    s3.auth = "default"
  }
}

webservice {
  port = 4053
  akka.http.server.request-timeout = 60s
}

docker {
  hash-lookup {
    enabled = false
  }
}

system {
  cromwell_id = "main"

  cromwell_id_random_suffix = false

  abort-jobs-on-terminate=true

  abort {
    # How frequently Cromwell should scan for aborts.
    scan-frequency: 30 seconds

    cache {
      # Guava cache concurrency.
      concurrency: 1
      # How long entries in the cache should live from the time they are added to the cache.
      ttl: 20 minutes
      # Maximum number of entries in the cache.
      size: 100000
    }
  }

  workflow-heartbeats {
    ttl = 10 minutes
    heartbeat-interval = 30 seconds
    write-failure-shutdown-duration = 5 minutes
    write-batch-size = 100
    write-threshold = 100
  }

  max-workflow-launch-count = 50
  max-concurrent-workflows = 5000
  new-workflow-poll-rate = 2

  io {
     number-of-requests = 100000
     per = 100 seconds
     number-of-attempts = 5
  }
}

call-caching {
  enabled = false
}

workflow-options {
    workflow-log-dir = "cromwell-workflow-logs"
    workflow-log-temporary = false
    workflow-failure-mode = ContinueWhilePossible
}

backend {
  default = "AWSBATCH"
  providers {
    AWSBATCH {
      actor-factory = "cromwell.backend.impl.aws.AwsBatchBackendLifecycleActorFactory"
      config {
        // Base bucket for workflow executions
        root = "<s3_bucket>"

        filesystems {
          s3 {
            // A reference to a potentially different auth for manipulating files via engine functions.
            auth = "default"
          }
        }

        // A reference to an auth defined in the `aws` stanza at the top.  This auth is used to create
        // Jobs and manipulate auth JSONs.
        auth = "default"

        numSubmitAttempts = 6
        numCreateDefinitionAttempts = 6

        default-runtime-attributes {
          queueArn: "<my_arn>
          scriptBucketName: "scripts"
        }
      }
    }
  }
}
```

Alternately, has anyone looked at this fork? Is it worth looking into?
https://github.com/henriqueribeiro/cromwell

**Comment by PeterT2020** [comment]
Just a quick note, the documentation for version 59 doesn't show anything about scriptBucketName anywhere, so I'm unsure why it is even being asked about:
https://github.com/broadinstitute/cromwell/blob/59/docs/tutorials/AwsBatch101.md

But when I didn't have scriptBucketName in my conf, I saw this instead:
2024-08-15 00:24:58,849  WARN  - Unrecognized configuration key(s) for AwsBatch: auth, numCreateDefinitionAttempts, numSubmitAttempts
2024-08-15 00:24:58,878 cromwell-system-akka.dispatchers.engine-dispatcher-83 ERROR - scriptBucketName is required
java.lang.RuntimeException: scriptBucketName is required

So I guess it is required?

**Reply by PeterT2020** [reply]
I have also tried it with the latest version now and it still didn't recognize those configuration items. Is the AWSBatch backend just completely broken this repo?

**Reply by aednichols** [reply, marked as answer]
Yeah, I definitely recommend the fork if you're using AWs.

**Comment by PeterT2020** [comment]
Resolution == use the aws-friendly fork, but even that is going to have some issues
